doc/sphinx/addendum/implicit-coercions.rst

```diff
@@ -82,6 +82,36 @@ To coerce an object :g:`t:C t₁..tₙ` of ``C`` towards ``D``, we have to
 apply the coercion ``f`` to it; the obtained term :g:`f _.._ t` is
 then an object of ``D``.
 
+Reverse Coercions
+-----------------
+
+A term is reverse-coerced only if it cannot be coerced (directly) to its expect type.
+A coercion is said to be reversible if it is declared with with the :attr:`reverse` attribute.
+
+In order to cast a term :g:`a : A` to type :g:`B` using a reverse coercion we:
+
+1. find a reversible coercion :g:`f : B >-> A`
+
+2. synthesize  some :g:`?x : B` such that :g:`f ?x = a`,
+   typically through :ref:`canonicalstructures` or :ref:`typeclasses`
+
+3. replace :g:`a` with the value of :g:`?x`
```

Comment:
Probably could be bullets.  We don't use numbered lists very much in the doc.  And perhaps it could just be a long sentence--it's not a series of steps the user must follow, e.g. "Coq finds ..., then synthesizes ... and finally replaces ..."